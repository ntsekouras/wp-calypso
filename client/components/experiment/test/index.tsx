/**
 * @jest-environment jsdom
 */

/**
 * External Dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';

import { Variation, DefaultVariation, RawExperiment } from '../index';

describe( 'Experiment Component', () => {
	const testComponent = ( variation?: string ) => (
		<RawExperiment name={ 'test' } variation={ variation }>
			<DefaultVariation name={ 'a' }>Default</DefaultVariation>
			<Variation name={ 'b' }>Variation B</Variation>
		</RawExperiment>
	);

	test( 'renders a default variation', () => {
		const { container } = render( testComponent( undefined ) );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders a default variation by name', () => {
		const { container } = render( testComponent( 'a' ) );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders a variation by name', () => {
		const { container } = render( testComponent( 'b' ) );
		expect( container ).toMatchSnapshot();
	} );
} );
