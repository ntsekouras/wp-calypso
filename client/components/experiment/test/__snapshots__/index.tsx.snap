// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiment Component renders a default variation 1`] = `
<div>
  Default
</div>
`;

exports[`Experiment Component renders a default variation by name 1`] = `
<div>
  Default
</div>
`;

exports[`Experiment Component renders a variation by name 1`] = `
<div>
  Variation B
</div>
`;
